### PR TITLE
Fixing #852 - Removing duplicate icons on the map viewer and defaulti…

### PR DESCRIPTION
**Pull request by rm-star**
> Thank you for submitting a pull request! Please add below details so that we can merge it faster.

---
#### Agreements

> Please check these boxes to ensure you've actually read and understood these important rules!
- [\* ] **I have reviewed and accepted the [guidelines for contributing](https://github.com/CellularPrivacy/Android-IMSI-Catcher-Detector/blob/development/.github/CONTRIBUTING.md) to this project.**
- [ _] *_I have reviewed and closely followed the [Style Guide](https://github.com/CellularPrivacy/Android-IMSI-Catcher-Detector/wiki/Style-Guide) for this Android app.**

---
#### Overview

Adding a fix for issue 852.
Removing duplicate user icons on the map by clearing out previous layers on a refresh and only creating new layers if a previous one does not already exist.
#### Classification

> Please click the correct checkbox that applies best to classify your pull request.
- [ *] Bugfix (non-breaking change which fixes an existing issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (restructuring of existing code without changing its external functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---
#### References

> If your pull request is related to or solves any existing Issues, please link them here.
> # 852
> ---
#### Screenshots

> If you have created visual changes, please add a screenshot without metadata here.
> N/A
> …ng to view the users current position instead of an indeterminate position

### CODE REVIEW

File: AIMSICD/src/main/java/com/secupwn/aimsicd/ui/fragments/MapFragment.java [hunk 1]
```diff
@@ -289,28 +291,37 @@ private void setupMapType(int mapType) {
...
+        mMap.getOverlayManager().clear();
         // Check if we were successful in obtaining the map.
         mMap.setBuiltInZoomControls(true);
         mMap.setMultiTouchControls(true);
         mMap.setMinZoomLevel(3);
         mMap.setMaxZoomLevel(19); // Latest OSM can go to 21!
         mMap.getTileProvider().createTileCache();
-        mCompassOverlay = new CompassOverlay(getActivity(), new InternalCompassOrientationProvider(getActivity()), mMap);
+
+        if (null == mCompassOverlay) {
```
Comment (mimi89999) [inline AIMSICD/src/main/java/com/secupwn/aimsicd/ui/fragments/MapFragment.java]: Could you keep the `null` and other values on the right of logical operators to keep the code consistent?

Comment (rm-star) [reply AIMSICD/src/main/java/com/secupwn/aimsicd/ui/fragments/MapFragment.java]: Certainly, it's done now.

**Comment by mimi89999** [comment]
In MapViewerOsmDroid.java also...

**Comment by rm-star** [comment]
Good catch, fixed now.

File: AIMSICD/src/main/java/com/secupwn/aimsicd/ui/fragments/MapFragment.java (hunk 1 above)
Comment (rm-star) [reply AIMSICD/src/main/java/com/secupwn/aimsicd/ui/fragments/MapFragment.java]: Fixed MapViewerOsmDroid.java as well

**Comment by mimi89999** [comment]
@rm-star The code looks OK. Could you squash your 3 commits into 1?

@larsgrefer Could you have a look at this PR?

**Review by mimi89999 — CHANGES_REQUESTED** [review]
The commits need to be squashed.

**Comment by rm-star** [comment]
I think it's squashed now

**Pull request merged**